rfcs/text/0025-container-metric-fields.md

```diff
@@ -88,6 +89,246 @@ Stage 2: Included a real world example source document. Ideally this example com
+    "cpu": {
+      "time": "2021-12-20T09:53:30.156Z",
+      "usageNanoCores": 18691146,
+      "usageCoreNanoSeconds": 4189523881380
+    },
+    "memory": {
+      "time": "2021-12-20T09:53:30.156Z",
+      "availableBytes": 1768316928,
+      "usageBytes": 2764943360,
+      "workingSetBytes": 2111090688,
+      "rssBytes": 2150400,
+      "pageFaults": 131567,
+      "majorPageFaults": 103
+    },
+    "network": {
+      "time": "2021-12-20T09:53:30.156Z",
+      "rxBytes": 1115133198,
+      "rxErrors": 0,
+      "txBytes": 812729002,
+      "txErrors": 0
+    },
+    "fs": {
+      "availableBytes": 98727014400,
+      "capacityBytes": 101258067968,
+      "usedBytes": 2514276352,
+      "inodesFree": 18446744073709551615,
+      "inodes": 6258720,
+      "inodesUsed": 138624
+    },
+    "runtime": {
+      "imageFs": {
+        "availableBytes": 98727014400,
+        "capacityBytes": 101258067968,
+        "usedBytes": 860204379,
+        "inodesFree": 18446744073709551615,
+        "inodes": 6258720,
+        "inodesUsed": 138624
+      }
+    }
+  },
+  "pods": [
+    {
+      "podRef": {
+        "name": "nginx-deployment-2303442956-pcqfc",
+        "namespace": "default",
+        "uid": "beabc196-2456-11e7-a3ad-42010a840235"
+      },
+      "startTime": "2021-12-20T09:53:30.156Z",
+      "containers": [
+        {
+          "name": "nginx",
+          "startTime": "2017-04-18T16:47:44Z",
+          "cpu": {
+            "time": "2017-04-20T08:06:34Z",
+            "usageNanoCores": 11263994,
+            "usageCoreNanoSeconds": 43959424
+          },
+          "memory": {
+            "time": "2017-04-20T08:06:34Z",
+            "usageBytes": 1462272,
+            "workingSetBytes": 1454080,
+            "rssBytes": 1409024,
+            "pageFaults": 841,
+            "majorPageFaults": 0
+          },
+          "rootfs": {
+            "availableBytes": 98727014400,
+            "capacityBytes": 101258067968,
+            "usedBytes": 61440,
+            "inodesFree": 6120096,
+            "inodes": 6258720,
+            "inodesUsed": 21
+          },
+          "logs": {
+            "availableBytes": 98727014400,
+            "capacityBytes": 101258067968,
+            "usedBytes": 28672,
+            "inodesFree": 6120096,
+            "inodes": 6258720,
+            "inodesUsed": 138624
+          },
+          "userDefinedMetrics": null
+        }
+      ],
+      "network": {
+        "time": "2021-12-20T09:53:30.156Z",
+        "rxBytes": 107056,
+        "rxErrors": 0,
+        "txBytes": 72447,
+        "txErrors": 0
+      },
+      "volume": [
+        {
+          "availableBytes": 1939689472,
+          "capacityBytes": 1939701760,
+          "usedBytes": 12288,
+          "inodesFree": 473551,
+          "inodes": 473560,
+          "inodesUsed": 9,
+          "name": "default-token-sg8x5"
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### Mapped event
+
+```json
+{
+  "container": {
```

Comment:
ECS describes a `host.*` as a computing instance including `hardware, virtual machines, Docker containers, and Kubernetes nodes`.

As I revisit this proposal, I'm worried about causing confusion. The existing `container.*` fields hold data only relevant to containers that wouldn't apply to `host.*`.

What's the motivation for duplicating the `cpu`, `memory`, and `network` fields underneath `container.*` in addition to `host.*`?

Reply:
With those new fields we are focusing on cpu, memory and network metrics from containers. In container orchestrators, host relates to the node(vm, bare metal) where the containers are running and these metrics are not the same as the container metrics. For example `kubernetes node` has its own cpu usage different to each container/pod running on it. So to me it is not a duplication, rather more container specific metrics.  Also I read the [sources](https://github.com/elastic/ecs/blob/main/rfcs/text/0005-host-metric-fields.md#source-data) of `host.*` ECS fields and it does not mention metrics coming from containers.  Maybe I am missing something. @kaiyan-sheng any thoughts on this?

Reply:
@MichaelKatsoulis That's my understanding too, I don't consider containers as a part of the host definition. @jsoriano WDYT?

Reply:
I will do it.

Reply:
Concerns updated!